srHook/SAPHanaSrTakeoverBlocker.py

```diff
@@ -85,69 +69,70 @@ def preTakeover(self, isForce, **kwargs):
                *    Sudoers does allow the query of the attribute (alternatively add <sid>adm user(s) to hacluster group)
             """
             method = "preTakeover"
+            start_time = time.time()
             self.tracer.info("{0}.{1}() called with isForce={2}".format(self.__class__.__name__, method, isForce))
             if not isForce:
                 # run pre takeover code
                 # run pre-check, return != 0 in case of error => will abort takeover
                 # for test purposes just block all sr_takeover() calls
-                tmpFile = tempfile.NamedTemporaryFile(prefix='SAPHanaSR_', suffix='_TBSRHOOK')
-                cmdOut = tmpFile.name
-                tmpFile.close()
-                mySID = os.environ.get('SAPSYSTEMNAME')
-                mysid = mySID.lower()
-                myAttribute = "hana_{0}_sra".format(mysid)
-                myCMD = "sudo /usr/sbin/crm_attribute -n {0} -G -t reboot -q".format(myAttribute)
-                self.tracer.info("{0}.{1}() myCMD is: {2}, cmdOut is: {3}".format(self.__class__.__name__, method, myCMD, cmdOut))
-                mySRA = ""
-                rc = os.system(myCMD + " > " + cmdOut)
-                if rc != 0:
-                    # srtakeover attribute not found
-                    return 0
+                tmp_file = tempfile.NamedTemporaryFile(prefix='SAPHanaSR_', suffix='_TBSRHOOK')
+                cmd_out = tmp_file.name
+                tmp_file.close()
+                my_sid = os.environ.get('SAPSYSTEMNAME')
+                low_sid = my_sid.lower()
+                my_attribute = "hana_{0}_sra".format(low_sid)
+                my_cmd = "sudo /usr/sbin/crm_attribute -n {0} -G -t reboot -q".format(my_attribute)
```

Comment:
Just an idea - could SAPHanaSR-SRHhelper also cover the query for the SRA-Attribute. This would give us:
a) One fork less directly in the python hook
b) Code of the hook could be simplifed
return code of SAPHanaSR-SRHhelper could be e.g. 4 or 6 to differ from maintenance.

Reply:
I do not think we need to check always both cluster attributes. SAPHanaSR-hookHelper could return, if the SRA attribute is set. And only check for the maintenance attribute, if this is not the case.
We can definitive do that later (if we want to do that). So *nothing* we need to do *now*.